cirq-core/cirq/transformers/transformer_primitives.py

```diff
@@ -426,3 +426,18 @@ def unroll_circuit_op_greedy_frontier(
         )
         frontier = unrolled_circuit.insert_at_frontier(sub_circuit.all_operations(), idx, frontier)
     return _to_target_circuit_type(unrolled_circuit, circuit)
+
+
+def xor_ops_with_tags(circuit: CIRCUIT_TYPE, tags: Sequence[Hashable], *, deep: bool = False):
```

Comment:
Please add a function docstring.

Also, I'm not sure about this name - we're not xor'ing ops, we're xor'ing their tags. Maybe (as used in the test) "flip_tags" or "toggle_tags" would be clearer?

Reply:
Yeah, naming this method is hard. changed to `toggle_tags` and added a docstring